File: micro-digiservices/ecreditservice/src/main/java/io/digiservices/ecreditservice/dto/DemandeModificationDto.java
package io.digiservices.ecreditservice.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DemandeModificationDto {
    private Integer demandeCreditId;
    private Long managerUserId;
    private Long agentUserId;
    private String motif;
    private String justification;
    private String priorite;
    private String dateLimite;
    private String commentaireManager;
    private List<ElementModificationDto> elements;
}

File: frontend/ultima-ng-19.0.0/src/app/interface/demande.modification.dto.ts
interface DemandeModificationDto {
    demandeCreditId: number;
    managerUserId: number;
    agentUserId: number;
    motif: string;
    justification?: string;
    priorite: 'URGENTE' | 'HAUTE' | 'NORMALE' | 'BASSE';
    dateLimite?: string;
    commentaireManager?: string;
    elements: ElementModification[];
}
